### PR TITLE
Speed up pre-commit checks

### PR DESCRIPTION
**Pull request by jonahkagan**
_Review by commits_

(╯°□°）╯︵ ┻━┻

A variety of tweaks found in various internet sources. Brings average pre-commit hook time down from >1m to <10s on my machine.

### CODE REVIEW

File: client/tsconfig.json [hunk 1]
```diff
@@ -12,9 +12,13 @@
...
+    "incremental": true,
+    "declaration": true,
+    "emitDeclarationOnly": true,
+    "outDir": "dist",
     "types": ["node", "@types/jest", "@testing-library/jest-dom"],
     "jsx": "preserve"
   },
-  "include": ["src"]
+  "include": ["src"],
+  "exclude": ["**/node_modules", "**/.*/"]
```
Comment (jonahkagan) [inline client/tsconfig.json:23]: Yes, we were type checking all of node_modules before 🤦

Comment (arsalansufi) [reply client/tsconfig.json:23]: 🙈

**Review by arsalansufi — APPROVED** [review]
LGTM!

Side note: I actually didn't even notice that we had pre-commit hooks set up in Arlo. They don't seem to be running for me 🤔

**Comment by jonahkagan** [comment]
@arsalansufi woah! Maybe we don't need them if you've been getting by fine without

**Comment by arsalansufi** [comment]
Yeah good point! Auto-formatting basically covers this need for me

File: client/tsconfig.json (hunk 1 above)
Comment (eventualbuddha) [reply client/tsconfig.json:23]: What, really? I'd expect TypeScript to know not to do this.

Comment (jonahkagan) [reply client/tsconfig.json:23]: Well, I'm not 100% sure actually. I think I remember adding that exclude line and rerunning the lint-staged command and seeing a big speedup. I got the tip from this doc: https://github.com/microsoft/TypeScript/wiki/Performance#misconfigured-include-and-exclude.

But reading closer in this other section of the doc, it says node_modules is excluded by default: https://github.com/microsoft/TypeScript/wiki/Performance#specifying-files. So maybe I was wrong or not diligent in only changing one setting at a time.

**Pull request merged**